modules/mod_ginger_adlib_elasticsearch2/mod_ginger_adlib_elasticsearch2.erl

```diff
@@ -0,0 +1,70 @@
+%% @doc Zotonic/Adlib integration
+-module(mod_ginger_adlib_elasticsearch2).
+-author("Driebit <tech@driebit.nl>").
+
+-mod_title("Adlib in ElasticSearch").
+-mod_prio(500).
+-mod_description("Makes Adlib data searchable by indexing it in ElasticSearch (v7+)").
+-mod_depends([mod_elasticsearch2, mod_ginger_adlib]).
+-mod_schema(1).
+
+-export([
+    init/1,
+    index/1,
+    types/1,
+    manage_schema/2,
+    observe_adlib_update/2
+]).
+
+-include_lib("mod_ginger_adlib/include/ginger_adlib.hrl").
+-include_lib("zotonic.hrl").
+
+init(Context) ->
+    default_config(index, index(Context), Context),
+    prepare_elasticsearch_index(Context).
+
+manage_schema(_, Context) ->
+    prepare_elasticsearch_index(Context).
+
+%% @doc Create index and mappings for each type in the index.
+-spec prepare_elasticsearch_index(z:context()) -> ok.
+prepare_elasticsearch_index(Context) ->
+    {Version, Mapping} = ginger_adlib_elasticsearch2_index_mapping:default_mapping(Context),
+    {ok, _} = elasticsearch2_index:upgrade(index(Context), Mapping, Version, Context),
+    ok.
+
+observe_adlib_update(#adlib_update{date = _Date, database = Database, record = #{<<"@attributes">> := #{<<"priref">> := Priref}} = Record}, Context) ->
+    lager:debug("Indexing Adlib record ~s from database ~s", [Priref, Database]),
+    MappedRecord = ginger_adlib_elasticsearch2_mapper:map(Record, ginger_adlib_elasticsearch_mapping),
```

Comment:
Is it ok that the module name passed is without the `2` suffix?

Reply:
Good one - that should have the `2`